internal/app/machined/internal/phase/platform/platform.go

```diff
@@ -25,6 +40,101 @@ func (task *Platform) RuntimeFunc(mode runtime.Mode) phase.RuntimeFunc {
 }
 
 func (task *Platform) runtime(platform platform.Platform, data *userdata.UserData) (err error) {
+	switch platform.Mode() {
```

Comment:
This file is the gist of things. I plan on cleaning this up a bit more but wanted to put the PR up so we can all talk about the general direction some time this week.